src/main/java/com/google/api/codegen/java/JavaGapicCodePathMapper.java

```diff
@@ -0,0 +1,81 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.java;
+
+import com.google.api.codegen.config.ProductConfig;
+import com.google.api.codegen.gapic.GapicCodePathMapper;
+import com.google.api.codegen.transformer.java.JavaSurfaceNamer;
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+import java.util.ArrayList;
+
+/**
+ * An implementation of GapicCodePathMapper that generates the output path from a prefix, and/or
+ * package name.
+ */
+@AutoValue
+public abstract class JavaGapicCodePathMapper implements GapicCodePathMapper {
+
+  private final String PACKAGE_SPLITTER = "\\.";
+
+  public abstract String prefix();
+
+  @Override
+  public String getOutputPath(String elementFullName, ProductConfig config) {
+    return getOutputPath(config, null);
+  }
+
+  @Override
+  public String getSamplesOutputPath(String elementFullName, ProductConfig config, String method) {
+    return getOutputPath(config, method);
+  }
+
+  private String getOutputPath(ProductConfig config, String methodSample) {
```

Comment:
you're only using the presence of `methodSample`, so how about s/`String`/`boolean`/ ?

Reply:
This and above:
We append the method name to the base package name for other languages (and in the original design doc I think). This is not in conflict with that the samples are in the same package. E.g.,
The two samples will be in the same package, but with two different class paths:
```
samples/src/main/java/com/google/cloud/examples/language/v1/snippets/analyzeentities/AnalyzeEntities.java
samples/src/main/java/com/google/cloud/examples/language/v1/snippets/analyzesentiment/AnalyzeSentiment.java
```

You could still invoke the samples by doing 
```
java com.google.cloud.example.language.v1.snippets.methodname.classname -arg1 -arg2
```
If we want to keep this behavior, we will keep `methodSample` here. WDYT?